runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkTranslationContext.java

```diff
@@ -68,4 +69,9 @@ public void computeOutputs() {
       dataset.action(); // force computation.
     }
   }
+
+  /** Generate a unique pCollection id number to identify runner-generated sinks. */
+  public int getSinkId() {
```

Comment:
nitpick: can you call this `nextSinkId` otherwise it could be confusing because it will be like a sort of getters with 'side effect'

Reply:
Done